receiver/googlecloudpubsubreceiver/internal/common_protos.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudpubsubreceiver/internal"
+
+import (
+	_ "google.golang.org/genproto/googleapis/cloud/audit"
+)
```

Comment:
I'd prefer to replace or augment this with functionality I'm hoping to propose in the builder, which allows adding go packages to be imported for side effects. this would let users add types they want accurate conversion for.

that is implemented here: https://github.com/kamalmarhubi/opentelemetry-collector/commit/d846d4d91ab1c78ff4e2e49811eca001ac453dcc

I initially wanted that for an approach I was considering for custom ottl functions: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16098#issuecomment-1553888863

but it would help here too, I think?